rdt/transformers/numerical.py

```diff
@@ -487,3 +488,142 @@ def _reverse_transform(self, data):
             data = self._univariate.ppf(scipy.stats.norm.cdf(data))
 
         return super()._reverse_transform(data)
+
+
+class BayesGMMTransformer(NumericalTransformer):
+    """Bayesian GMM transformer."""
+
+    DETERMINISTIC_TRANSFORM = False
+    DETERMINISTIC_REVERSE = True
+    COMPOSITION_IS_IDENTITY = True
+
+    def __init__(self, dtype=None, nan='mean', null_column=None, max_clusters=10,
+                 weight_threshold=0.005):
+        super().__init__(dtype=dtype, nan=nan, null_column=null_column)
+        self._max_clusters = max_clusters
+        self._weight_threshold = weight_threshold
+        self._output_dimensions = None
+        self._column_raw_dtypes = None
+        self._bgm_transformer = None
+        self._valid_component_indicator = None
+
+    def get_output_types(self):
+        """Return the output types supported by the transformer.
+
+        Returns:
+            dict:
+                Mapping from the transformed column names to supported data types.
+        """
+        output_types = {
+            'continuous': 'float',
+            'discrete': 'categorical'
+        }
+        if self.null_transformer and self.null_transformer.creates_null_column():
+            output_types['is_null'] = 'float'
+
+        return self._add_prefix(output_types)
+
+    def _fit(self, data):
+        """Fit the transformer to the data.
+
+        Args:
+            data (pandas.Series):
+                Data to fit to.
+        """
+        self._bgm_transformer = BayesianGaussianMixture(
+            n_components=self._max_clusters,
+            weight_concentration_prior_type='dirichlet_process',
+            weight_concentration_prior=0.001,
+            n_init=1
+        )
+
+        self._bgm_transformer.fit(data.array.reshape(-1, 1))
+        self._valid_component_indicator = self._bgm_transformer.weights_ > self._weight_threshold
+        self._output_dimensions = 1 + self._valid_component_indicator.sum()
+
+        data = pd.DataFrame(data)
+        self._column_raw_dtypes = data.infer_objects().dtypes
+
+    def _transform_helper(self, data):
+        data = data.reshape((len(data), 1))
+        means = self._bgm_transformer.means_.reshape((1, self._max_clusters))
+        stds = np.sqrt(self._bgm_transformer.covariances_).reshape((1, self._max_clusters))
+        normalized_values = ((data - means) / (4 * stds))[:, self._valid_component_indicator]
+        component_probs = self._bgm_transformer.predict_proba(data)
+        component_probs = component_probs[:, self._valid_component_indicator]
+
+        num_components = self._valid_component_indicator.sum()
+        selected_component = np.zeros(len(data), dtype='int')
+        for i in range(len(data)):
+            component_porb_t = component_probs[i] + 1e-6
+            component_porb_t = component_porb_t / component_porb_t.sum()
+            selected_component[i] = np.random.choice(np.arange(num_components), p=component_porb_t)
```

Comment:
Is it possible to redo this section as a vector operation? Something like
```
component_probs = (component_probs + 1e-6) / component_probs.sum()
np.random.choice(np.arange(num_components), p=component_probs)
```

Reply:
The problem is that `p` in `np.random.choice` needs to be 1-D. While there are work arounds (I didn't find any alternative functions that accept multi-dimensional `p`, but we could always write our own), I'm not sure if they are worth it.